### ISSUE TITLE
Lantern does DNS resolutions in order to reach loggly and google analytics

### ISSUE THREAD

**Issue by xiam** [getlantern/lantern#2482]
Captured some DNS resolutions while Lantern was idle:

![screen shot 2015-05-06 at 1 16 14 pm](https://cloud.githubusercontent.com/assets/385670/7500239/1c49977c-f3f2-11e4-8f4a-0978f26adfd4.png)

**Comment by myleshorton** [comment]
We shouldn't actually be resolving the DNS for analytics or loggly at all since both of them are tunneled (along with everything else theoretically) so we the DNS shouldn't be resolved until the server side is reached.

**Comment by fffw** [comment]
Loggly is not in the proxied site list, same for pure-journey-3547.herokuapp.com. We should add them.

google-analytics.com is in the list. Theoretically it should not be resolved directly. Are there any possibility that they came from other applications?

**Comment by myleshorton** [comment]
Oh that makes sense @fffw. We used to not need them on the list since we were just configuring things internally to go through a proxy, but with detour we do. Analytics is also at least attempting do do direct domain fronting, which seems incorrect to me and could be related to a DNS lookup leak. I'll keep looking at this.

**Comment by xiam** [comment]
Analytics is no longer showing up but loggly remains.

![screen shot 2015-06-02 at 4 54 42 pm](https://cloud.githubusercontent.com/assets/385670/7948252/246501c4-0948-11e5-8b2b-c2f59fd5b426.png)

I'm capturing traffic using a VM with just few services enabled so there is no chance that other app besides Lantern is trying to reach loggly.

**Comment by fffw** [comment]
Yeah I can find accessing to logs-01.loggly.com directly from my logs, I think that's because my existing lantern.yaml doesn't contain loggly.com in proxied sites. Once we run `genconfig` next time, the changes to `proxiedsites/original.txt` will be propagated I think.

**Comment by xiam** [comment]
Let me try again with a clean lantern.yaml

**Comment by xiam** [comment]
Tested again but no luck guys:

![screen shot 2015-06-03 at 6 41 03 pm](https://cloud.githubusercontent.com/assets/385670/7973863/22ce1e26-0a20-11e5-8115-15832a243918.png)

And in fact google analytics came again after some time.

This is the current valencia branch I only used Lantern and cURL to carry on the test, so there is no risk other program is doing these requests. 

I already removed the `.lantern` directory and I'm going to continue digging into this one, any other thoughts?

**Comment by xiam** [comment]
I found this only happens when detour is called and it makes sense since when the address is not whitelisted Lantern [tries to connect directly](https://github.com/getlantern/lantern/blob/2b574098d84528e38d8cc8c521e78b2e159143ea/src/github.com/getlantern/detour/detour.go#L120).

So for some reason loggly and google analytics trigger detour, but update.getlantern.org does not.

@fffw Do you know how we can avoid loggly and google analytics from triggering this detour lookup?

**Comment by fffw** [comment]
As the changes of https://github.com/getlantern/lantern/pull/2545 didn't put into production yet, could you try add them directly to proxiedsites of lantern.yaml during your test? @xiam 

I think update.getlantern.org doesn't trigger detour because getlantern.org is already whitelisted.

**Comment by fffw** [comment]
I tried triggering error reporting on beta 5 running at an Ubuntu 14.10 VM. DNS query for `logs-01.loggly.com` could be spotted at first, but disappeared after if I added `loggly.com` to proxied sites.

**Comment by myleshorton** [comment]
I already put it in the proxied sites list in 156f0dec0bb40fddfc0a3d9bade0b6d91d0cd4df

Did you do something different?

**Comment by fffw** [comment]
@myleshorton You just put it into `proxiedsites/original.txt`. It will not take effect until we run `genconfig` again.

**Comment by myleshorton** [comment]
Oh excellent point @fffw thanks!

**Comment by xiam** [comment]
@fffw Sorry, [#2545](https://github.com/getlantern/lantern/pull/2545) was already merged into valencia so I supposed that was the only thing we needed, should I also run genconfig like this?

```
cd ./src/github.com/getlantern/flashlight/genconfig
./genconfig.bash
```

**Comment by xiam** [comment]
After running `genconfig.bash` the fallbacks map gets build like this:

```
var fallbacks = map[string]*client.ChainedServerInfo{
  "fallback-178.62.239.34": &client.ChainedServerInfo{
    Addr:      "178.62.239.34:443",
    Cert:      "-----BEGIN CERTIFICATE-----\nMIIFPjCCAyagAwIBAgIEU1H3pTANBgkqhkiG9w0BAQsFADBHMQswCQYDVQQGEwJVUzEQMA4GA1UE\nCBMHRmxvcmlkYTEUMBIGA1UEBxMLQWNjb3VudGFibGUxEDAOBgNVBAM
    AuthToken: "ub0YJccdngnxqfM0fiaOb6dtxiiVRIFhoxleKyfUYijfUiuUh81BaRR5N5Xgsaio",
    Pipelined: true,
    Weight:    1000000,
    QOS:       10,
    trusted:   true,
  },
...
```

And when I try to build Lantern I have lots of errors:

```
# github.com/getlantern/flashlight/config
src/github.com/getlantern/flashlight/config/fallbacks.go:15: unknown client.ChainedServerInfo field 'trusted' in struct literal
src/github.com/getlantern/flashlight/config/fallbacks.go:24: unknown client.ChainedServerInfo field 'trusted' in struct literal
```

The field that `ChainedServerInfo` expects is `Trusted`. I can manually replace `trusted` with `Trusted` but it would be better to fix that one too.

**Comment by myleshorton** [comment]
That's really just a bug in https://github.com/getlantern/lantern/blob/valencia/src/github.com/getlantern/flashlight/genconfig/fallbacks.go.tmpl -- trusted should be capitalized there, and then I believe it will work.

**Comment by myleshorton** [comment]
One interesting aspect of that @xiam is that it means the Trusted field was never set before for our ChainedServerInfos, which meant that prior to fetching the config over the network none of those would be considered trusted. We definitely need to fix that for beta 6.

**Comment by xiam** [comment]
Awesome, thanks @myleshorton! My tests came out clean now. I'm going to let the proxy run for a bit more just to be absolutely sure we're not leaking any other DNS request.

**Comment by xiam** [comment]
> We definitely need to fix that for beta 6.

Yes, definitely!

**Comment by myleshorton** [comment]
Sweet thanks @xiam!

**Comment by xiam** [comment]
Hey @fffw, could you help me testing this one? https://github.com/getlantern/lantern/pull/2611

Thanks!

**Comment by fffw** [comment]
@xiam Yeah both sites can be added to yaml, and Wireshark no long spots DNS query to them.
